app/clients/letter/dvla.py

```diff
@@ -80,5 +103,99 @@ def authenticate(self):
 
         return response.json()["id-token"]
 
-    def send_letter(self):
-        pass
+    def _get_auth_headers(self):
+        return {
+            "Accept": "application/json",
+            "Authorization": self.jwt_token,
+            "X-API-Key": self.dvla_api_key.get(),
+        }
+
+    def send_letter(
+        self,
+        notification_id: str,
+        address: list[str],
+        postage: str,
+        service_id: str,
+        organisation_id: str,
+        pdf_file: bytes,
+    ):
+        """
+        Sends a letter to the DVLA for printing
+
+        address should be normalised address lines, e.g. ['A. User', 'London', 'SW1 1AA']
+
+        normalised address lines can be returned by calling:
+        `PostalAddress.from_personalisation(notification.personalisation).normalised_lines`
+        """
+        from app.models import INTERNATIONAL_POSTAGE_TYPES
+
+        if postage in INTERNATIONAL_POSTAGE_TYPES:
+            raise NotImplementedError
+
+        current_app.logger.info(f"Sending letter with id {notification_id}")
+
+        try:
+            response = requests.post(
+                f"{current_app.config['DVLA_API_BASE_URL']}/print-request/v1/print/jobs",
+                headers=self._get_auth_headers(),
+                json=self.format_create_print_job_json(
+                    notification_id, address, postage, service_id, organisation_id, pdf_file
+                ),
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            # Catch errors and raise our own to indicate what action to take.
+            # If the error has details, we add them to the error message.
+            if e.response.status_code == 400:
+                raise DvlaException(response.json()["errors"][0]["detail"])
+            elif e.response.status_code in {401, 403}:
+                raise DvlaUnauthorisedRequestException(
+                    f'{e.response.status_code} {response.json()["errors"][0]["detail"]}'
+                )
+            elif e.response.status_code == 409:
+                raise DvlaDuplicatePrintRequestException(response.json()["errors"][0]["detail"])
+            elif e.response.status_code == 429:
+                raise DvlaThrottlingException(str(e))
+            elif e.response.status_code >= 500:
+                raise DvlaRetryableExcepton(str(e))
+            else:
+                raise DvlaException(str(e))
+        else:
+            return response.json()
+
+    def format_create_print_job_json(
+        self, notification_id, address_lines, postage, service_id, organisation_id, pdf_file
+    ):
+        from app.models import FIRST_CLASS
+
+        recipient_name = address_lines.pop(0)
```

Comment:
Commented below, but again this is a mutation of incoming data that probably isn't ideal. Anything that calls `send_letter` will now have different data than they might expect.